src/AuthContext.js

```diff
@@ -0,0 +1,80 @@
+import React from "react";
+import produce from "immer";
+
+import apiClient from "./lib/apiClient";
+
+const AuthContext = React.createContext();
+
+class AuthProvider extends React.Component {
+  constructor(props) {
+    super(props);
+    
+    this.state = {
+      checkingAuthentication: true,
+      isAuthenticated: false
+    };
+  }
+
+  componentWillMount() {
+    const authToken = localStorage.getItem("cah-token");
+
+    if (authToken) {
+      apiClient("GET", "/auth")
```

Comment:
I think the readability of this function could possibly be improved. You may consider creating a `removeAuth` function that encapsulates the following logic: 
```
     if (authToken) { 
          localStorage.removeItem("cah-token");
     }

     this.setState(
       produce(draft => {
          draft.checkingAuthentication = false;
          })
       );
```
Also consider removing the `else` statement for the first `if(authToken)` statement and convert to an early exit statement. 
```
if (!authToken) {
   this.removeAuth()
}
```